src/pyodide-py/pyodide/_core.py

```diff
@@ -1,44 +1,10 @@
-import platform
-from typing import Any, Callable
+import sys
 
-if platform.system() == "Emscripten":
-    from _pyodide_core import (
-        JsProxy,
-        JsException,
-        create_proxy,
-        create_once_callable,
-    )
-else:
-    # Can add shims here if we are so inclined.
-    class JsException(Exception):  # type: ignore
-        """
-        A wrapper around a Javascript Error to allow the Error to be thrown in Python.
-        """
+if "_pyodide_core" not in sys.modules:
+    from _pyodide import _core as _pyodide_core
```

Comment:
Maybe,
```suggestion
    from _pyodide_inner import _core as _pyodide_core
```
to make it clear that it's really not a module the user would be interested in. The difference between pyodide and _pyodide is not that clear.

Reply:
I think it's a standard thing for Python packages to have private versions with a leading underscore. For instance pytest does this:
https://github.com/pytest-dev/pytest/tree/6.2.2/src